src/python/pants/backend/helm/subsystems/helm.py

```diff
@@ -171,6 +171,12 @@ class HelmSubsystem(TemplatedExternalTool):
         advanced=True,
     )
 
+    infer_docker_image_dependencies = BoolOption(
+        default=True,
+        help="If true, parse k8s manifests generated by helm to find docker image references.",
```

Comment:
should we include a hint on why you'd want to use this? Maybe "Disabling this option and manually specifying dependencies can improve performance"

Reply:
I would suggest to move this option to the `helm-infer` subsystem and potentially rename the option to be `deployment_dependencies`. 

The `helm-infer` subsystem was created to alter the behavior of the dependency inference, so it makes sense to me to be there. And the option name suggestion is because it feels to me more targeted at disabling the inference of deployment dependencies that simply Docker images (even though Docker images are the only type of dependencies inferred).

That last one is quite nick picky I know, it just feels more intentionally to me from an end user perspective. Not a blocker though.